authentication/auth.go

```diff
@@ -461,50 +662,82 @@ func GetRequiredMetadata(launcherSchema surveys.LauncherSchema) ([]Metadata, str
 
 	log.Println("Loading metadata from schema:", url)
 
+	var schema QuestionnaireSchema
 	resp, err := clients.GetHTTPClient().Get(url)
 	if err != nil {
 		log.Println("Failed to load schema from:", url)
-		return nil, fmt.Sprintf("Failed to load Schema from %s", url)
+		return schema, fmt.Sprintf("Failed to load Schema from %s", url)
 	}
 
 	if resp.StatusCode != 200 {
 		log.Print("Invalid response code for schema from: ", url)
-		return nil, fmt.Sprintf("Failed to load Schema from %s", url)
+		return schema, fmt.Sprintf("Failed to load Schema from %s", url)
 	}
 
 	responseBody, err := ioutil.ReadAll(resp.Body)
 	resp.Body.Close()
 	if err != nil {
 		log.Print(err)
-		return nil, fmt.Sprintf("Failed to load Schema from %s", url)
+		return schema, fmt.Sprintf("Failed to load Schema from %s", url)
 	}
 
-	var schema QuestionnaireSchema
 	if err := json.Unmarshal(responseBody, &schema); err != nil {
 		log.Print(err)
-		return nil, fmt.Sprintf("Failed to unmarshal Schema from %s", url)
+		return schema, fmt.Sprintf("Failed to unmarshal Schema from %s", url)
 	}
 
-	defaults := GetDefaultValues()
+	return schema, ""
+}
 
-	for i, value := range schema.Metadata {
-		schema.Metadata[i].Default = defaults[value.Name]
+func getMandatatoryClaims(surveyType string, defaults map[string]string) []Metadata {
+	claims := make([]Metadata, 0)
+	if surveyType == "health" || surveyType == "social" {
+		claims = []Metadata{
+			{"questionnaire_id", "false", defaults["questionnaire_id"]},
+			{"case_ref", "false", defaults["case_ref"]},
+		}
 
-		if value.Validator == "boolean" {
-			schema.Metadata[i].Default = "false"
+	} else {
+		claims = []Metadata{
+			{"ru_ref", "false", defaults["ru_ref"]},
+			{"period_id", "false", defaults["period_id"]},
+			{"user_id", "false", defaults["user_id"]},
 		}
 	}
 
-	return schema.Metadata, ""
+	return claims
+}
+
+func getMissingMandatoryClaims(claims []string, mandatoryClaims []Metadata) []Metadata {
+	missingClaims := make([]Metadata, 0)
+	for _, v := range mandatoryClaims {
+		if !(stringInSlice(v.Name, claims)) {
+			missingClaims = append(missingClaims, v)
+		}
+	}
+
+	return missingClaims
+}
+
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
 }
 
 // GetDefaultValues Returns a map of default values for metadata keys
 func GetDefaultValues() map[string]string {
-
 	defaults := make(map[string]string)
-
 	collectionExerciseSid, _ := uuid.NewV4()
+	questionnaireId, _ := uuid.NewV4()
```

Comment:
Questionnaire ID needs to be a 16 digit string. The validation for this is correct in the schemas repo. The md doc uses UUID but as per validation, this will need to be a 16 digit string.